File: tests/test_class_lib.py
"""Library with class-based code for testing class method observation."""


class Calculator:
    def add(self, a, b):
        return {"result": a + b, "operation": "add"}

    def multiply(self, a, b):
        return {"result": a * b, "operation": "multiply"}

    def square(self, x):
        return {"result": x * x, "input": x}


class Formatter:
    def format_name(self, first, last):
        return {"display": f"{first} {last}", "first": first, "last": last}

    def format_currency(self, amount, currency):
        return {"formatted": f"{currency}{amount:.2f}", "amount": amount, "currency": currency}

File: tests/test-class-lib.js
/**
 * Library with class-based code for testing class method observation.
 */

class Calculator {
  add(a, b) {
    return { result: a + b, operation: 'add' };
  }

  multiply(a, b) {
    return { result: a * b, operation: 'multiply' };
  }

  square(x) {
    return { result: x * x, input: x };
  }
}

class Formatter {
  formatName(first, last) {
    return { display: `${first} ${last}`, first, last };
  }

  formatCurrency(amount, currency) {
    return { formatted: `${currency}${amount.toFixed(2)}`, amount, currency };
  }
}

module.exports = { Calculator, Formatter };
